app/webapp/src/shared/types/types.posts.ts

```diff
@@ -7,6 +7,22 @@ import { PlatformPost } from './types.platform.posts';
  */
 export interface GenericPostData {
   content: string;
+  metadata: {
+    originalPlatformAuthor: {
```

Comment:
I would create a `GenericPostAuthor` interface which will be the author object we associate to any GenericPost, and will be the interface we use to talk about content and authors with the parser.

What about this interface?

```ts
export interface GenericPostAuthor {
  platformId: PLATFORM;
  id: string;
  username: string;
  name: string;
}

export interface GenericPostData {
  url?: string;
  author: GenericPostAuthor;
  content: string;
  quotedPosts?: GenericPostData[];
}

```